src/turbomind/kernels/sampling_topp_kernels.h

```diff
@@ -19,135 +19,62 @@
-                             cudaStream_t    stream,
-                             cudaDeviceProp* cuda_device_prop,
-                             const bool*     skip_decode);
-
-template<typename T>
-void invokeAddBiasSoftMax(T*           logits,
-                          const T*     bias,
-                          const int*   end_ids,
-                          const bool*  finished,
-                          const int    m,
-                          const int    n_padded,
-                          const int    n,
-                          cudaStream_t stream);
-
-namespace segmented_topp_impl {
-enum DType_t
-{
-    kFLOAT,
-    kHALF,
-    kINT8
-};
-
-template<typename Key_Data_Type_   = float,
-         typename Value_Data_Type_ = int32_t,
-         int BLOCK_THREADS_        = 256,
-         int KEYS_PER_LDG_         = 1>
-struct Segmented_topk_kernel_params {
-    typedef Key_Data_Type_   Key_Data_Type;
-    typedef Value_Data_Type_ Value_Data_Type;
-    enum
+struct BlockPrefixCallbackOp {
+    // Running prefix
+    float running_total;
+    // Constructor
+    __device__ BlockPrefixCallbackOp(float running_total): running_total(running_total) {}
+    // Callback operator to be entered by the first warp of threads in the block.
+    // Thread-0 is responsible for returning a value for seeding the block-wide scan.
+    __device__ float operator()(float block_aggregate)
     {
-        BLOCK_THREADS = BLOCK_THREADS_
-    };
-    enum
-    {
-        ITEMS_INCREMENT = 32
-    };
-    // enum { KEYS_PER_LDG = 2 * 4 / sizeof(Key_Data_Type_) };
-    enum
-    {
-        KEYS_PER_LDG = KEYS_PER_LDG_
-    };
+        float old_prefix = running_total;
+        running_total += block_aggregate;
+        return old_prefix;
+    }
 };
 
-struct TopKPerSegmentContext {
-    TopKPerSegmentContext(): sm_count(0), sm_shared_size(0), sm_version(0){};
-    int sm_count;
-    int sm_shared_size;
-    int sm_version;
+struct TopPSortParams {
+    void*  workspace{};
+    size_t workspace_size{};
+    void*  logits;
+    void*  sorted_logits;
+    int*   sorted_indices;
+    int*   kept;
+    int*   top_ks;
+    float* top_ps;
+    int    batch_size;
+    int    vocab_size;
+    int    vocab_size_padded;
 };
 
-struct TopKPerSegmentParams {
-    // input/output keys and values
-    void *gmem_src_keys, *gmem_dst_keys, *gmem_dst_vals;
-    // not used in the custom implementation
-    void* gmem_src_vals;
-    // int array of size num_segments
-    int* gmem_active_count_per_segment;
-    int* gmem_active_count_total;
-    int* gmem_begin_offsets;
-    // gmem_end_offsets will be populated
-    int*  gmem_end_offsets;
-    void* workspace;
-    // total number of items for all segments
-    int num_items;
-    int num_segments;
-    // top_k per segment
-    int   num_top_k;
-    float top_p;
-    float confidence_threshold;
+template<typename T>
+void invokeTopPSort(TopPSortParams& params, cudaStream_t stream);
+
+struct TopPMinPFilterParams {
```

Comment:
`struct TopPMinPFilterParams` is a subset of `struct TopPSortPrams` except the `float* min_ps`. 
Can we merge them together?